boot/BootFromDevice.c

```diff
@@ -0,0 +1,155 @@
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ *   Copyright (C) 2019 Stanislav Motylkov                                 *
```

Comment:
Significant portions of the code here come from LoadLinux.c, namely `BootLoadFlashCD`, we should try to respect the copyright of that author here somehow

Reply:
Added `Copyright (C) 2004-2006 Xbox Linux authors`.